### PR TITLE
Fixes #923 other text value retained even when other not selected.

### LINKED ISSUE #923: Other text field retains data even the user changes the option
### Expected behaviour
Must wipe the data entered in other text field before when user changes the option.

### Actual behaviour
Retains previously entered text.

### Steps to reproduce the behaviour
Launch any survey which has an other text field option
select other and enter data and click save and continue
navigate back using previous change the option and click save and continue
navigate back and select the other option again

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
Fixes #923.

### How to review 
All automated tests should pass.
Open either `test_checkbox.json` and `test_radio.json` schemas.
Select the `Other` option and enter a value into the text field.
Save and continue.
Go back.
Select a different option.
Save and continue.
Go back.
Select the `Other` option again.
Confirm that the previous value is not in the Other text field.

Should also work when selecting multiple options (checkbox).

### CODE REVIEW

File: app/forms/questionnaire_form.py [hunk 1]
```diff
@@ -93,8 +93,23 @@ def answer_errors(self, input_id):
...
+        clear_other_text_field(block_json, data, form_data)
+
+        form = QuestionnaireForm(form_data, meta={'csrf': False})
     else:
         form = QuestionnaireForm(meta={'csrf': False})
 
     return form
+
+
+def clear_other_text_field(block_json, data, form_data):
```
Comment (iwootten) [inline app/forms/questionnaire_form.py]: At the moment, the form_helper has some deserialisation logic around dates within it. I would think that this clearing of the other text field would be better placed alongside as the data wouldn't be manipulated as part of the form logic then.

File: app/forms/questionnaire_form.py [hunk 2]
```diff
@@ -93,8 +93,23 @@ def answer_errors(self, input_id):
...
+    for question in SchemaHelper.get_questions_for_block(block_json):
+        for answer in question['answers']:
+            if 'parent_answer_id' in answer and \
+                    answer['parent_answer_id'] in data and \
+                    'Other' not in form_data.getlist(answer['parent_answer_id']) and \
+                    form_data.get(answer['id']) is not None and \
```
Comment (iwootten) [inline app/forms/questionnaire_form.py]: I think these last two conditions can simply be expressed as:

    and form_data.get(answer['id'])

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/forms/questionnaire_form.py (hunk 1 above)
Comment (iwootten) [inline app/forms/questionnaire_form.py]: Actually, this clear could possibly be achieved within the get_answer_fields method?

**Review by jonnyshaw89 — APPROVED** [review]

**Review by iwootten — APPROVED** [review]

**Comment by samiwel** [comment]
Squashed and rebased ready to be merged.

**Pull request merged**